backend/app/controllers/spree/admin/resource_controller.rb

```diff
@@ -134,7 +134,16 @@ def belongs_to(model_name, options = {})
     end
   end
 
+  def product_exists?
+    return unless parent.nil?
+
+    flash[:error] = flash_message_for(Spree::Product.new, :not_found)
+    redirect_to admin_products_path
+  end
+
   def resource_not_found
+    return product_exists? if model_class == Spree::Image
```

Comment:
This seems like we should be making these changes to `Spree::Admin::ImagesController` rather than making a conditional like this in the generic base resource controller.